api/app/weather_models/process_grib.py

```diff
@@ -183,6 +242,47 @@ def store_bounding_values(self,
         session.add(prediction)
         session.commit()
 
+    def process_env_can_grib_file(self, session: Session, dataset, grib_info: ModelRunInfo,
+                                  prediction_run: PredictionModelRunTimestamp):
+        # for GDPS, RDPS, HRDPS models, always only ever 1 raster band in the dataset
+        raster_band = dataset.GetRasterBand(1)
+        # Iterate through stations:
+        for (points, values) in self.yield_data_for_stations(raster_band):
+            self.store_bounding_values(
+                points, values, prediction_run, grib_info, session)
+
+    def process_noaa_grib_file(self, session: Session, dataset, grib_info: ModelRunInfo,
+                               prediction_run: PredictionModelRunTimestamp):
+        precip_raster_band = None
+        if (grib_info.model_run_timestamp - grib_info.prediction_timestamp).total_seconds() == 0:
+            tmp_raster_band = dataset.GetRasterBand(GFS_000_HOURS_RASTER_BANDS.get('tmp_tgl_2'))
+            rh_raster_band = dataset.GetRasterBand(GFS_000_HOURS_RASTER_BANDS.get('rh_tgl_2'))
+            u_wind_raster_band = dataset.GetRasterBand(GFS_000_HOURS_RASTER_BANDS.get('u_comp_wind_10m'))
+            v_wind_raster_band = dataset.GetRasterBand(GFS_000_HOURS_RASTER_BANDS.get('v_comp_wind_10m'))
+        elif (grib_info.prediction_timestamp - grib_info.model_run_timestamp).total_seconds() > 0:
+            tmp_raster_band = dataset.GetRasterBand(GFS_003_HOURS_RASTER_BANDS.get('tmp_tgl_2'))
+            rh_raster_band = dataset.GetRasterBand(GFS_003_HOURS_RASTER_BANDS.get('rh_tgl_2'))
+            u_wind_raster_band = dataset.GetRasterBand(GFS_003_HOURS_RASTER_BANDS.get('u_comp_wind_10m'))
+            v_wind_raster_band = dataset.GetRasterBand(GFS_003_HOURS_RASTER_BANDS.get('v_comp_wind_10m'))
+            precip_raster_band = dataset.GetRasterBand(GFS_003_HOURS_RASTER_BANDS.get('apcp_sfc_0'))
+
+        for (points, values) in self.yield_data_for_stations(tmp_raster_band):
```

Comment:
Lots of repetition here with these for blocks, I think a parameterized method could be pulled out here to reduce the duplication.